### PR TITLE
fix(UI): replace button with icons in challenge lower jaw

### LINKED ISSUE #47157: Replace button with icons in the lower jar of the challenges
### Is your feature request related to a problem? Please describe.

As discussed in #47037, swapping buttons with icons then adding tool tips for the icons, will open the possibility for more features that can be helpful for campers

### Describe the solution you'd like

Adding icons and their tooltips to lower jaw with a helping page, that replace the buttons

### Describe alternatives you've considered

Not skilled enough to think of one

### Additional context

We have to make sure that these icons and its tool tips are accessible as mentioned by @bbsmooth, and @ahmadabdolsaheb. For additional info please read their discussion on https://github.com/freeCodeCamp/freeCodeCamp/issues/47037

And I would really love to have the reset icon show after one attempt, but we can give it time in the forum first see if learners complain

### PR DESCRIPTION
**Pull request by mehdibouaziz**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #47157

- replaced the text inside help and reset buttons with icons
- added title and aria-label for tooltip - the tooltip uses the translate function to have a localised text
- removed the btn-block class to allow a more compact UI which will allow more icons in the future

### CODE REVIEW

**Comment by ahmadabdolsaheb** [comment]
Solid work here. Here are a few things that need to be addressed before this gets merged.

1) Are the icons your design. If so, please look in client/src/assets/icons for custom svg samples and usage. If not, what is the license?
2) Please add data-cy to the icon buttons and select the reset button accordingly from the related cypress test so the tests pass
3) Could you right align the buttons for consistency with mobile's new learn UI.

**Review by alicalimli — COMMENTED** [review]
Nice

**Comment by mehdibouaziz** [comment]
@ahmadabdolsaheb
I modified accordingly but I could find the proper test in cypress, I don't think the tests for these two buttons were written yet. The previous buttons I replaced were also missing data-cy.
I'm not familiar with writting cypress tests so maybe it needs a new issue ?

**Review by Sboonny — COMMENTED** [review]
I can't review it currently, but I am wondering if refactoring the code like below is possible?

**Comment by nayabatir1** [comment]
> Hi @Sboonny, thanks for the remarks! I used {condition ? : : <></>} instead of {condition && } because I read some time ago that the ternary version should be prefered to avoid bugs, after struggling with a bug myself. This [article](https://medium.com/geekculture/stop-using-for-conditional-rendering-in-react-a0f7b96200f8) discusses the issue but suggest to use {condition ? : : null} instead. I'll let you decide which solution you prefer :)
> 
> The missing () around the condition is an error on my part, I'll fix it in the next commit.

Hey @mehdibouaziz, I read the article you shared, there is no harm in using short circuit expression, you just have to make sure that the condition evaluates to boolean and not truthly/falsy.

**Comment by mehdibouaziz** [comment]
@ahmadabdolsaheb  I found the selector error in the failing cypress test and put the correct one.
@nayabatir1 Hi, thanks for your feedback. True, but I tried with the short circuit and in this case one of the conditions as it was written evaluates to 0 and causes a 0 to render:
image
fcc-conditionnal-0
I switched to ternary with `null` just to be sure :)

**Comment by nayabatir1** [comment]
You can refer to this article https://www.codingem.com/javascript-double-exclamation-operator/

File: client/src/templates/Challenges/classic/lower-jaw.css
Comment (ahmadabdolsaheb) [inline client/src/templates/Challenges/classic/lower-jaw.css:13]: They could live in the editor.css similar to other lower jaw styles.

File: client/src/templates/Challenges/classic/lower-jaw.tsx
Comment (ahmadabdolsaheb) [inline client/src/templates/Challenges/classic/lower-jaw.tsx:203]: Both custom SVGs should be stand alone react components and placed in client/src/assets/icons.

Feel free to look at other icon component samples.

**Review by ahmaxed — COMMENTED** [review]
This for making the modifications. It should be ready to merge after these changes.

File: client/src/templates/Challenges/classic/lower-jaw.css
Comment (mehdibouaziz) [reply client/src/templates/Challenges/classic/lower-jaw.css]: done: moved to editor.css and increased margin to 10px

File: client/src/templates/Challenges/classic/lower-jaw.tsx
Comment (mehdibouaziz) [reply client/src/templates/Challenges/classic/lower-jaw.tsx]: done for both icons

**Comment by mehdibouaziz** [comment]
All done :)

note that I tried to add parenthesis to the condition for the conditional display of the button, but when committing prettier removes them, they are unnecessary I suppose.
` {isAttemptsLargerThanTest && !challengeIsCompleted ? (`

**Comment by bbsmooth** [comment]
Just a minor nitpick. Do we need to float the get help button to the left of the reset button? Usually the tab sequence goes left to right so it seems a little odd to have it go right to left for these two buttons. This isn't technically an accessibility failure though, so I'm not demanding anything here. Although I would highly recommend that we add our own custom focus outline on these buttons (instead of relying on the browser's default, which isn't always so great) so that it is perfectly clear which one has focus. I've been using the outline color `--blue-mid` for things I work on because it has accessible contrast with both our light and dark themes. But I understand that that is not necessarily an official policy.

**Comment by mehdibouaziz** [comment]
Hi, thanks for the review!

You mean that with the current solution when using tab the focus is moving from right to left because of float:right ?
I can switch to a flexbox instead if it would improve accessibility, there's already a container bar for the buttons.

Since the help button is hidden at first and appears after a few fails, I thought it would be better to have it appear on the left so it doesn't shuffle around the reset button.

For the focus color I'll let the owners decide, it's easy to add.

**Comment by bbsmooth** [comment]
No need to switch to flexbox for accessibility, this is purely about DOM order of element, which flexbox won't change. Although if we eventually add more buttons then I think flexbox would be the better solution here.

Another option might be to place the buttons on the left side instead of the right. Then when we add the help button it would be placed to the right of the reset button and wouldn't cause a shift and the tab order would follow the DOM order.

**Comment by ahmadabdolsaheb** [comment]
Nice catch @bbsmooth, I just simplified the styles and the buttons should show up on the left side of the button jaw in the correct DOM order.

@mehdibouaziz, once the tests pass LGMT.

**Review by ahmaxed — APPROVED** [review]

**Review by ahmaxed — APPROVED** [review]

**Review by Sboonny — CHANGES_REQUESTED** [review]
nitpick, when someone hovers over the button, the tooltip can block the visual indicator of the other button.

Should we position the tooltip, a little to the top, instead of the bottom of the courser? Or will this affect introduce some sort of CSS bug 🤔 

image

**Review by Sboonny — APPROVED** [review]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @mehdibouaziz, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Pull request merged**
